integrationTests/singleShard/block/executingMiniblocksSc_test.go

```diff
@@ -0,0 +1,342 @@
+	withdrawValue := big.NewInt(10)
+	mintAllNodes(nodes, initialVal)
+
+	deployScTx(nodes, idxProposer, string(scCode))
+	proposeBlock(nodes, idxProposer, round)
+	syncBlock(t, nodes, idxProposer, round)
+	round = incrementAndPrintRound(round)
+
+	nodeDoesJoinGame(nodes, idxProposer, topUpValue, hardCodedScResultingAddress)
+	proposeBlock(nodes, idxProposer, round)
+	syncBlock(t, nodes, idxProposer, round)
+	round = incrementAndPrintRound(round)
+
+	checkJoinGameIsDoneCorrectly(
+		t,
+		nodes,
+		idxProposer,
+		idxProposer,
+		initialVal,
+		topUpValue,
+		hardCodedScResultingAddress,
+	)
+
+	nodeCallsRewardAndSend(nodes, idxProposer, idxProposer, withdrawValue, hardCodedScResultingAddress)
+	proposeBlock(nodes, idxProposer, round)
+	syncBlock(t, nodes, idxProposer, round)
+	round = incrementAndPrintRound(round)
+
+	proposeBlock(nodes, idxProposer, round)
+	syncBlock(t, nodes, idxProposer, round)
+	round = incrementAndPrintRound(round)
+
+	checkRewardIsDoneCorrectly(
+		t,
+		nodes,
+		idxProposer,
+		idxProposer,
+		initialVal,
+		topUpValue,
+		withdrawValue,
+		hardCodedScResultingAddress,
+	)
+
+	checkRootHashes(t, nodes, []int{0})
+
+	time.Sleep(1 * time.Second)
+}
+
+func incrementAndPrintRound(round uint32) uint32 {
+	round++
+	fmt.Printf("#################################### ROUND %d BEGINS ####################################\n\n", round)
+
+	return round
+}
+
+func mintAllNodes(nodes []*integrationTests.TestProcessorNode, value *big.Int) {
+	for _, n := range nodes {
+		if n.ShardCoordinator.SelfId() == sharding.MetachainShardId {
+			continue
+		}
+
+		for _, n2 := range nodes {
+			integrationTests.MintAddress(n.AccntState, n2.PkTxSignBytes, value)
+		}
+	}
+}
+
+func deployScTx(nodes []*integrationTests.TestProcessorNode, senderIdx int, scCode string) {
+	fmt.Println("Deploying SC...")
+	txDeploy := createTxDeploy(nodes[senderIdx], scCode)
+	nodes[senderIdx].SendTransaction(txDeploy)
+	fmt.Println("Delaying for disseminating the deploy tx...")
+	time.Sleep(stepDelay)
+
+	fmt.Println(integrationTests.MakeDisplayTable(nodes))
+}
+
+func proposeBlock(nodes []*integrationTests.TestProcessorNode, idxProposer int, round uint32) {
+	fmt.Println("Proposing block...")
+	for idx, n := range nodes {
+		if idx != idxProposer {
+			continue
+		}
+
+		body, header := n.ProposeBlock(round)
+		n.BroadcastAndCommit(body, header)
+	}
+
+	fmt.Println("Delaying for disseminating headers and miniblocks...")
+	time.Sleep(stepDelay)
+	fmt.Println(integrationTests.MakeDisplayTable(nodes))
+}
+
+func syncBlock(t *testing.T, nodes []*integrationTests.TestProcessorNode, idxProposer int, round uint32) {
+	fmt.Println("All other shard nodes sync the proposed block...")
+	for idx, n := range nodes {
+		if idx == idxProposer {
+			continue
+		}
+
+		err := n.SyncNode(uint64(round))
+		if err != nil {
+			assert.Fail(t, err.Error())
+			return
+		}
+	}
+
+	time.Sleep(stepDelay)
+	fmt.Println(integrationTests.MakeDisplayTable(nodes))
+}
+
+func nodeDoesJoinGame(
```

Comment:
I would name it `nodeJoinsGame`, when I first saw a call on it I thought it was something boolean, like "does it join the game?". It would be easier to read the test

Reply:
good one, changed. Also in multishard/block